platforms/core-configuration/kotlin-dsl/src/test/kotlin/org/gradle/kotlin/dsl/ProjectExtensionsTest.kt

```diff
@@ -5,20 +5,18 @@ import com.nhaarman.mockito_kotlin.doReturn
 import com.nhaarman.mockito_kotlin.eq
 import com.nhaarman.mockito_kotlin.inOrder
 import com.nhaarman.mockito_kotlin.mock
-import com.nhaarman.mockito_kotlin.times
 import com.nhaarman.mockito_kotlin.whenever
 import org.gradle.api.Action
 import org.gradle.api.NamedDomainObjectContainer
 import org.gradle.api.NamedDomainObjectFactory
 import org.gradle.api.Project
 import org.gradle.api.UnknownDomainObjectException
-import org.gradle.api.reflect.TypeOf
 import org.junit.Assert.fail
 import org.junit.Test
 
 
 @Suppress("deprecation")
-class ProjectExtensionsTest {
+class ProjectExtensionsTest { // TODO: review and act
```

Comment:
The important part of this test is asserting on the behavior when extensions are not found.

Before removing conventions we had to do some gymnastic, leniently try to get the extension, if not found try to get the convention and if not found then forcibly getting the extension to propagate the right exception.

Now with conventions removed we can simply get the extension and the extension container will throw the right exception.

Reply:
Please take a look at my last commit.

Reply:
Hmm, see the cross version test failures... Aren't we overdoing the simplification?

Reply:
The inlined functions with reified generics get, well, inlined.
We should rework the test to only exercise access to extensions now.